src/DataGrid.tsx

```diff
@@ -908,10 +935,18 @@ function DataGrid<R, SR, K extends Key>(
   }
 
   function getLayoutCssVars() {
-    if (autoResizeColumn === null) return layoutCssVars;
+    if (autoResizeColumn === null && flexWidthViewportColumns.length === 0) return layoutCssVars;
     const { gridTemplateColumns } = layoutCssVars;
     const newSizes = gridTemplateColumns.split(' ');
-    newSizes[autoResizeColumn.idx] = 'max-content';
+    if (autoResizeColumn !== null) {
+      newSizes[autoResizeColumn.idx] = 'max-content';
+    }
+    if (flexWidthViewportColumns.length > 0) {
+      for (const column of flexWidthViewportColumns) {
+        newSizes[column.idx] = column.width.toString();
```

Comment:
Is `toString()` necessary? Can we assert that it's a string already?
```suggestion
        newSizes[column.idx] = column.width as string;
```